crowbar_framework/app/models/service_object.rb

```diff
@@ -1183,6 +1187,14 @@ def apply_role(role, inst, in_queue)
       elems.each do |role_name|
```

Comment:
Can you add a `next if role_name =~ /_remove$/` here, just to be safe? Otherwise, if someone manually adds a _remove role in the element_order attribute, then we will have some fun stuff happening.